### PR TITLE
Added scan/verify as Scan/Verify in ASR

### PR DESCRIPTION
**Pull request by czgdp1807**
https://github.com/lfortran/lfortran/issues/758

Commit - 55f9929506ecb3c9aaf2082d2ad0cc33f3cc780a in https://gitlab.com/lfortran/lfortran/-/merge_requests/1754/commits

### CODE REVIEW

**Comment by certik** [comment]
Thanks for this, it looks good. Let's add some tests and merge.

**Comment by certik** [comment]
@czgdp1807 do you want to finish this? Is there anything else needed for stdlib?

**Comment by czgdp1807** [comment]
I will finish this. I was busy with some new features in LPython. 

> Is there anything else needed for stdlib?

The following commits are to be applied over LFortran main for stdlib to work (from https://gitlab.com/lfortran/lfortran/-/merge_requests/1754/commits)

1. 55f9929506ecb3c9aaf2082d2ad0cc33f3cc780a
2. 748ad26cbb29adad15052e4ab8c09c3b56b166a4
3. 2bcda964b7a713927c05fcf6a5566be4019cdfd0
4. 4b33cd67e2462e532c7608afc00933ba95019807

**Comment by czgdp1807** [comment]
@certik Ready for review. Please let me know if anything else is to be done here.

**Comment by certik** [comment]
This needs more discussion, so I am converting it to a Draft for now.

File: src/libasr/ASR.asdl [hunk 1]
```diff
@@ -295,6 +295,8 @@ expr
     | SetPop(expr a, ttype type, expr? value)
     | IntegerBitLen(expr a, ttype type, expr? value)
     | Ichar(expr arg, ttype type, expr? value)
+    | Scan(expr string, expr set, expr? back, ttype type, expr? value)
+    | Verify(expr string, expr set, expr? back, ttype type, expr? value)
```
[the remark this answers is not in the document]
Comment (czgdp1807) [reply src/libasr/ASR.asdl:299]: The return type of the `verify` depends on the input argument `kind`. Basically the return type of `verify` is decided at runtime. We need `verify` for compiling `fpm` as well, so we need to decide on this soon.

**Comment by czgdp1807** [comment]
The purpose of this PR is complete.

**Pull request closed without merging**